### ISSUE TITLE
False CA2000

### ISSUE THREAD

**Issue by jherby2k** [dotnet/roslyn-analyzers#2412]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Steps to Reproduce

Minimum repro (referencing Microsoft.PowerShell.SDK):

```c#
    public sealed class ModuleFixture : IDisposable
    {
        readonly Runspace _runspace;

        public ModuleFixture()
        {
            var state = InitialSessionState.CreateDefault();

            // This bypasses the execution policy (InitialSessionState.ExecutionPolicy isn't available with PowerShell 5)
            state.AuthorizationManager = new AuthorizationManager("Microsoft.PowerShell");

            _runspace = RunspaceFactory.CreateRunspace(state);
            _runspace.Open();
        }

        public void Dispose()
        {
            _runspace.Dispose();
        }
    }
```

**Expected Behavior**:

No warning. The runspace is disposed with the ModuleFixture.

**Actual Behavior**:

CA2000 warning on `RunspaceFactory.CreateRunspace(state)`.

**Comment by mavasani** [comment]
Verified repro on 2.9.2 package, but no repro on latest 2.9.3.x pre-release myget package for sources in master: https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.3-beta1.19252.1+136aefb3

Fix pending 2.9.3 release, closing.
